From fad5a09f019fcd58749d67a644e0521ae562947a Mon Sep 17 00:00:00 2001
From: Koji Ishii <kojii@chromium.org>
Date: Tue, 22 Oct 2024 03:49:38 +0000
Subject: [PATCH] Remove `NOTREACHED` in `InlineCursor::MoveTo(const LayoutObject&)`

Originally, `InlineCursor` expects callers to check if the
`LayoutObject` is OOF, and do what is appropriate for the
caller. For that reason, it had `NOTREACHED`.

After discussing at the crbug.com/371112526, AX prefers
`InlineCursor` to be the false state silently. This patch
removes the `NOTREACHED` for the change.

(cherry picked from commit 6c7f1b0051c295348e03f989f8a659159bcae429)

(cherry picked from commit d3cb3f3b44e4f979b13b386ec3db116e9d52e8ba)

Fixed: 371112526
Change-Id: Ied2e5cb4cf192d0d23db94905cb3056b7022a005
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5939013
Auto-Submit: Koji Ishii <kojii@chromium.org>
Reviewed-by: Kevin Ellis <kevers@chromium.org>
Commit-Queue: Kevin Ellis <kevers@chromium.org>
Cr-Original-Original-Commit-Position: refs/heads/main@{#1369555}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5943420
Auto-Submit: Alison Maher <almaher@microsoft.com>
Commit-Queue: Alison Maher <almaher@microsoft.com>
Commit-Queue: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Cr-Original-Commit-Position: refs/branch-heads/6778@{#246}
Cr-Original-Branched-From: b21671ca172dcfd1566d41a770b2808e7fa7cd88-refs/heads/main@{#1368529}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5949985
Reviewed-by: Koji Ishii <kojii@chromium.org>
Commit-Queue: Koji Ishii <kojii@chromium.org>
Cr-Commit-Position: refs/branch-heads/6723@{#1468}
Cr-Branched-From: 985f2961df230630f9cbd75bd6fe463009855a11-refs/heads/main@{#1356013}
---

--- a/third_party/blink/renderer/core/layout/inline/inline_cursor.cc
+++ b/third_party/blink/renderer/core/layout/inline/inline_cursor.cc
@@ -1425,7 +1425,6 @@
 void InlineCursor::MoveTo(const LayoutObject& layout_object) {
   DCHECK(layout_object.IsInLayoutNGInlineFormattingContext());
   if (layout_object.IsOutOfFlowPositioned()) [[unlikely]] {
-    NOTREACHED_IN_MIGRATION();
     MakeNull();
     return;
   }
--- a/third_party/blink/renderer/core/layout/inline/inline_cursor.h
+++ b/third_party/blink/renderer/core/layout/inline/inline_cursor.h
@@ -529,9 +529,10 @@
   // Functions to enumerate fragments for a |LayoutObject|.
   //
 
-  // Move to first |FragmentItem| or |NGPaintFragment| associated to
-  // |layout_object|. When |layout_object| has no associated fragments, this
-  // cursor points nothing.
+  // Move to the first `FragmentItem` associated to the `layout_object`.
+  // This cursor points nothing if any of the following conditions are true:
+  // * The `layout_object` has no associated fragments.
+  // * The `layout_object.IsOutOfFlowPositioned()`.
   void MoveTo(const LayoutObject& layout_object);
 
   // Same as |MoveTo|, except that this enumerates fragments for descendants
